Il2CppInterop.Runtime/Injection/ClassInjector.cs

```diff
@@ -583,10 +582,12 @@ private static bool IsMethodEligible(MethodInfo method)
         foreach (var parameter in method.GetParameters())
         {
             var parameterType = parameter.ParameterType;
-            if (!IsTypeSupported(parameterType))
+            if (!IsTypeSupported(parameterType) ||
+                method.IsStatic && parameterType.IsGenericType && parameterType.ContainsGenericParameters)
```

Comment:
Why does it matter here if the method is static or not?

Reply:
I mean Il2cpp Interop supports normal methods that are generic. If I don't include static part, those won't work. 

This check is here so that only non generic static methods are included.
Most likely I didn't want to bother supporting generic static methods.

Reply:
Do we support generic instance methods?